### DISCUSSION TITLE
Open an existing file on Create?

### DISCUSSION THREAD

**Discussion by drventure** [SilentVoid13/Templater#891] [Help]
I'm trying to get a small template working and can't quite get where I'd like to be.
I'm using Templater and Hotkeys for Templates.
I've assigned a Hotkey to create a new note in a specific folder using a Templater template. So far so good.

Next, I want to RENAME that new note with a datestamp
`<% tp.file.rename(tp.date.now("YYYY-MM-DD ddd")) _%>`
Again, so far so good.

Finally, IF a file by that datestamp name +already+ exists, I'd like to open it and jump to the end of the note automatically, and just not create any new note.
I +can+ determine whether the file exists or not
`	var fullname = tp.file.folder(true) + "/" + name + ".md"
	var exists = await tp.file.exists(fullname)`

But that's as far as I can get.
For the removing the newly created note, I've tried `tp.file.move(".trash")`, didn't work
`await app.vault.trash(file, true);`
but nothing I did for getting the proper "file" object or name seemed to work

For activating the existing note, similar issues.
I think it's something to do with 
`window.app.workspace.activeLeaf.view.editor`
but I'm not sure.

Any pointers?

**Comment by AB1908** [comment]
Can you check the Obsidian API? You're looking to create a new leaf and it should be under Workspace. Let me know if you run into trouble. <https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts>

**Comment by drventure** [comment]
I suspect you're right.

I found this poking around in the "Periodic Notes" plugin source
`const leaf = inNewSplit ? workspace.splitActiveLeaf() : workspace.getUnpinnedLeaf();`
`await leaf.openFile(file, { active: true });`

which definitely seems to be along the lines of what I need.

I'll keep digging. 

Thanks!

**Reply by drventure** [reply]
Ok. Well, I figured out something that works, although it's gotta be the nastiest way to go about it... If anyone cares to correct my awful coding here, I'd love to hear it!

The formatting in this post is going to be a little janky because I'm using obsidian execution commands to inject frontmatter or separators, but here it is. So far, seems pretty bulletproof, but it'd definitely not exhaustively tested yet.

==========================================================================
TEMPLATE STARTS BELOW

<%*

	//------------------------------------------------------------------------------
    //QUICKNOTE Template
    //Uses: Templater, Hotkeys for Templates (to make it easy to fire off the template)
    //General description:
    //Ive used Hotkeys for templates to assign this template to a specific folder 
    //via the Ctrl-Q key
	//
	//Then, I can hit Ctrl-Q from anywhere in Obsidian and one of two things will happen
    //I generate the name of the note using a standard date format
    //If a note with that name DOES NOT exist, then I use standard Templater functionallity
    //to create it and rename it to the desired name.
    //
	//If a note with that name DOES already exist, then open it, jump to the end of
	//the file, and insert a horizontal rule for separation and leave the cursor at then
	//end of file.
	//
	//++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	//IMPORTANT NOTE!
	//Do not use single quotes (apostrophies) in comments in this template as it 
	//messes up Templater and causes errors.
	//++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	//------------------------------------------------------------------------------
	//
	//get the normal daily template name, then append the current path
	var name = tp.date.now("YYYY-MM-DD ddd")
	var fullname = tp.file.folder(true) + "/" + name + ".md"
	var exists = await tp.file.exists(fullname)
	if (!exists) {
		//new note so insert frontmatter, format is wonky here, but it works
	 _%>
---
aliases: <% tp.file.creation_date("YYYY-MM-DD") %>
created: <% tp.file.creation_date("YYYY-MM-DD HH:mma") %>
---	
<%*

		//for new note, rename it after a bit of delay
		setTimeout(() => {  tp.file.rename(name) }, 500)
	} else {
		//there's an existing note
		var inNewSplit = false
		var workspace = window.app.workspace
		var file = window.app.vault.getAbstractFileByPath(fullname);
		var leaf = inNewSplit ? workspace.splitActiveLeaf() : workspace.getUnpinnedLeaf();
		await leaf.openFile(file, { active: true })
		leaf.view.editor.focus()
		leaf.view.editor.exec("goEnd")
		//var endch = leaf.view.editor.getLine(leaf.view.editor.lastLine).length
		//tR += endch
		//leaf.view.editor.replaceRange('', { line: 0, ch: 0 },{ line: 0, ch: endch })
		//output a separator 
		%>

---
<%*

		setTimeout(async () => {
			leaf.view.editor.focus()
			leaf.view.editor.exec("goEnd")
			file = app.vault.getAbstractFileByPath(tp.file.path(true))
			await app.vault.trash(file, true); 
		}, 500)
	}
_%>


==========================================================================
TEMPLATE ENDS ABOVE

**Reply by AB1908** [reply]
Very cool! If it works, well, you know the rest! 😄

**Reply by drventure** [reply]
It does appear to anyway. Thanks again for the pointer.

I'm more a .net guy that JS but once you know where to look, it's not too bad....
